tests/runner/test_dataset_block.py

```diff
@@ -131,107 +169,167 @@ def test_partial_block_for_chunked_data(
 def test_partial_block_for_chunked_data_with_padding_center(
     slicing_dim: Literal[0, 1, 2]
 ):
-    block_shape = [10, 10, 10]
-    block_shape[slicing_dim] = 6
-    start_index = 3
-    data = np.ones(block_shape, dtype=np.float32)
-    global_index = [0, 0, 0]
-    global_index[slicing_dim] = 10 + start_index
+    # sizes and shapes
     global_shape_t = [10, 10, 10]
     global_shape_t[slicing_dim] = 30
     global_shape = make_3d_shape_from_shape(global_shape_t)
-    chunk_shape_t = [10, 10, 10]
-    chunk_shape_t[slicing_dim] += 4  # for padding
-    chunk_shape = make_3d_shape_from_shape(chunk_shape_t)
+
+    unpadded_chunk_shape = (10, 10, 10)
+    padding = (2, 2)
+    padded_chunk_shape_t = list(unpadded_chunk_shape)
+    padded_chunk_shape_t[slicing_dim] += padding[0] + padding[1]
+    padded_chunk_shape = make_3d_shape_from_shape(padded_chunk_shape_t)
+
+    unpadded_block_slices = 2
+    unpadded_block_shape_t = list(unpadded_chunk_shape)
+    unpadded_block_shape_t[slicing_dim] = unpadded_block_slices
+    unpadded_block_shape = make_3d_shape_from_shape(unpadded_block_shape_t)
+    padded_block_shape_t = unpadded_block_shape_t
+    padded_block_shape_t[slicing_dim] += padding[0] + padding[1]
+    padded_block_shape = make_3d_shape_from_shape(padded_block_shape_t)
+    block_start_index = 3
+
+    chunk_start_index = 10  # start of chunk within global data, without padding
+
+    data = np.ones(padded_block_shape, dtype=np.float32)
     angles = np.linspace(0, math.pi, global_shape[0], dtype=np.float32)
     block = DataSetBlock(
         data=data,
         aux_data=AuxiliaryData(angles=angles),
         slicing_dim=slicing_dim,
-        block_start=start_index,
-        chunk_start=10,
+        block_start=block_start_index,
+        chunk_start=chunk_start_index,
         global_shape=global_shape,
-        chunk_shape=chunk_shape,
-        padding=(2, 2),
+        chunk_shape=padded_chunk_shape,
+        padding=padding,
```

Comment:
It could indeed deduce the padded chunk shape from the padding parameter. But then I would find it confusing when asking the block for its chunk_shape, as that would return something else then what's passed to the constructor... The way it is now is:

```python
block = DataSetBlock(..., chunk_shape=XXX, padding=..., ...)
assert block.chunk_shape == XXX
```

If we deduce the chunk shape with padding included from that parameters, it would look like this (pseudo-code):

```python
block = DataSetBlock(..., chunk_shape=XXX, padding=pad ...)
assert block.chunk_shape == XXX + pad
```

To me, the first part is more consistent.

The alternative of deducing padding from the chunk shape given and the block's shape won't work either, as we don't know how much of it is before and how much is after.

Reply:
In the constructor there's logic to handle when `global_shape` is passed in and padding needs to be accounted for:
```python
        if global_shape is None:
            global_shape_t = list(data.shape)
            global_shape_t[slicing_dim] -= padding[0] + padding[1]
            self._global_shape = make_3d_shape_from_shape(global_shape_t)
        else:
            self._global_shape = global_shape
```

Couldn't a similar thing be done using the `chunk_shape` and `padding` parameters of the constructor to set `self._chunk_shape` accordingly, so then the `self.chunk_shape` property still returns the same as it did before, just that the constructor sets it (rather than the creator of the block)?